x/hashlock/context_test.go

```diff
@@ -18,40 +17,45 @@ func TestContext(t *testing.T) {
 	random := weave.NewAddress(foo)
 
 	bg := context.Background()
-	cases := []struct {
+	cases := map[string]struct {
 		ctx   weave.Context
 		perms []weave.Condition
 		match []weave.Address
 		not   []weave.Address
 	}{
-		{bg, nil, nil, []weave.Address{sig, other, random}},
-		{
-			withPreimage(bg, foo),
-			[]weave.Condition{PreimageCondition(foo)},
-			[]weave.Address{other},
-			[]weave.Address{sig, random},
+		"empty context": {
+			ctx: bg,
+			not: []weave.Address{sig, other, random},
 		},
-		{
-			withPreimage(bg, []byte("one more time")),
-			[]weave.Condition{PreimageCondition([]byte("one more time"))},
-			nil,
-			[]weave.Address{sig, other, random},
+		"context with a preimage": {
```

Comment:
pp: please always start with the happy path in tests to make it easier to read and to understand the logic for reading the cases. Even if the execution is not necessarily sequential.